test/lib/rule-finder.js

```diff
@@ -139,15 +144,177 @@ const getRuleFinderForDedupeTests = proxyquire('../../src/lib/rule-finder', {
     '@noCallThru': true,
     '@global': true
   }
-});
+};
+
+const getRuleFinder = isV8Eslint 
+  ? proxyquire('../../src/lib/rule-finder', {
+      eslint: {
+        Linter: class {
+          getRules() {
+            return new Map()
+              .set('foo-rule', {})
+              .set('old-rule', {meta: {deprecated: true}})
+              .set('bar-rule', {})
+              .set('baz-rule', {});
+          }
+        }
+      },
+      module: {
+        createRequire: (relative) => mockCreateRequire(
+          () => getRuleFinder,
+          [
+            'eslint-plugin-plugin',
+            'eslint-plugin-no-rules',
+            '@scope/eslint-plugin-scoped-plugin',
+            '@scope/eslint-plugin',
+            '@scope-with-dash/eslint-plugin-scoped-with-dash-plugin',
+            '@scope-with-dash/eslint-plugin'
+          ],
+          relative
+        ),
+        '@global': true
+      },
+      ...mock
+    })
+  : (specifiedFileRelative, options) => {
+      const config = specifiedFileRelative 
+        ? proxyquire(path.resolve(specifiedFileRelative), mock)
+        : proxyquire(path.resolve(process.cwd(), require(path.join(process.cwd(), 'package.json')).main), mock);
+      
+      const ruleFinder = proxyquire('../../src/lib/rule-finder', {
+        eslint: {
+          // for v9 ESLint
+          ESLint: class {
+            // Mock to validate with rules that are not in ESLint Mock to validate with rules that are not in ESLint, 
+            //  or else you will get an error that the rule does not exist in ESLint
+            isPathIgnored(filePath) {
+              return filePath.includes('.ts') ? true : false;
+            }
+            // Mock to validate with rules that are not in ESLint Mock to validate with rules that are not in ESLint, 
+            //  or else you will get an error that the rule does not exist in ESLint
+            // eslint-disable-next-line no-unused-vars
+            calculateConfigForFile(filePath) {
+              const mergedConfig = {};
+              defaultConfig.forEach(config => {
+                merge(mergedConfig, config);
+              });
+
+              if(!specifiedFileRelative) {
+                if (Array.isArray(config)) {
+                  config.forEach(config => {
+                    merge(mergedConfig, config);
+                  });
+                  return mergedConfig;
+                } else {
+                  return merge(mergedConfig, config);
+                }
```

Comment:
similarly:
```suggestion
                return Object.assign(mergedConfig, ...[].concat(config || []));
```

same throughout